src/main/java/org/folio/rest/impl/PutOrdersByIdHelper.java

```diff
@@ -62,13 +54,40 @@
 
   }
 
+  private CompletableFuture<List<CompositePoLine>> changePoLinesReceiptStatus(CompositePurchaseOrder compPO, JsonObject poFromStorage) {
+    CompletableFuture<List<CompositePoLine>> compositePoLines;
+    if (isTransitionToOpen(compPO, poFromStorage)) {
+      if (isEmpty(compPO.getCompositePoLines())) {
+        return HelperUtils.getCompositePoLines(compPO.getId(), lang, httpClient, ctx, okapiHeaders, logger)
+          .thenApply(pols -> pols.stream().map(poline -> {
+            // Update receipt status from Pending to awaiting receipt
+            if (poline.getReceiptStatus() == CompositePoLine.ReceiptStatus.PENDING) {
+              poline.setReceiptStatus(CompositePoLine.ReceiptStatus.AWAITING_RECEIPT);
+            }
+            compPO.getCompositePoLines().add(poline);
+            return poline;
+          }).collect(Collectors.toList()));
+      } else {
+        compPO.getCompositePoLines().forEach(poLine -> {
+          if (poLine.getReceiptStatus() == CompositePoLine.ReceiptStatus.PENDING) {
+            poLine.setReceiptStatus(CompositePoLine.ReceiptStatus.AWAITING_RECEIPT);
+          }
+        });
+      }
+    }
+    compositePoLines = completedFuture(compPO.getCompositePoLines());
+
+    return compositePoLines;
+  }
+
   /**
    * Handles update of the order. First retrieve the PO from storage and depending on its content handle passed PO.
    */
   public void updateOrder(String orderId, CompositePurchaseOrder compPO) {
     compPO.setId(orderId);
     getPurchaseOrderById(orderId, lang, httpClient, ctx, okapiHeaders, logger)
       .thenCompose(poFromStorage -> validatePoNumber(compPO, poFromStorage))
+      .thenCompose(poFromStorage -> changePoLinesReceiptStatus(compPO, poFromStorage).thenApply(v -> poFromStorage))
```

Comment:
If an error occurs when the order is transferred to the "Open" status, it may happen that the receipt status will be changed, but order status will remain"Pending". It may be better to change the receipt status in `PutOrderLineByIdHelper.updateInventory()` method? In this case, do not need additional call to get poLines by order id.

Reply:
please, take a look at updated implementation